Detectors/Base/include/DetectorsBase/ContVec.h

```diff
@@ -0,0 +1,362 @@
+
+  /// set/get expand policy, see constructor documentation
+  void  setExpandPolicy(int val)       {getHeader()->expandPolicy = val;}
+  int   getExpandPolicy()        const {return getHeader()->expandPolicy;}
+
+  /// get number of objects stored
+  sizeType size()                const {return getHeader()->nObjects;}
+
+  /// get currently booked capacity
+  sizeType capacity()            const {return (sizeInBytes()-dataOffset())/sizeof(T);}
+
+  /// get i-th object pointer w/o boundary check
+  T*    operator[](sizeType i)   const {return getData()+i;}
+
+  /// get i-th object pointer with boundary check
+  T*    at(sizeType i)           const {return i<size() ? (*this)[i] : nullptr;}
+
+  /// get last object pointer
+  T*    back()                   const {return size() ? (*this)[size()-1] : nullptr;}
+
+  /// get 1st object pointer
+  T*    front()                  const {return size() ? (*this)[0] : nullptr;}
+  
+  /// add copy of existing object to the end, return created copy pointer
+  T*    push_back(const T& obj);
+
+  /// create an object with supplied arguments in the end of the container
+  template<typename ...Args> T* emplace_back(Args&&... args);
+  
+  /// clear content w/o changing capacity, if requested, explicitly delete objects
+  void  clear(bool calldestructor=true);
+
+  /// book space for objects and aux data
+  void  reserve(sizeType n=1000);
+
+  /// expand space for new objects
+  void  expand();
+
+  /// get raw pointer on the buffer, can be sent to another process
+  char*               getPtr()    const {return mPtr.get();}
+
+  /// returns a buffer pointer (releases the ownership) and reset the container
+  char*               release();
+
+  /// return total size in bytes
+  sizeType            sizeInBytes() const {return getHeader()->sizeInBytes;}
+
+  /// add objects stored in the container to vector<T> branch in the supplied tree
+  void AddToTree(TTree* tree, const std::string& brName);
+  
+ protected:
+
+  /// cast the head of container buffer to internal layout
+  Header*             getHeader() const {return reinterpret_cast<Header*>(getPtr());}
+  T*                  getData()   const {return reinterpret_cast<T*>(getPtr()+dataOffset());}
+
+  /// return next free slot, for "new with placement" creation, not LValue
+  T*                  nextFreeSlot();
+      
+ protected:
+
+  // offset of data wrt buffer start: account for Header + padding after Header block to respect T alignment
+  constexpr sizeType dataOffset() const {return sizeof(Header) + sizeof(Header)%alignof(T);};
+  // for some reason using static constexpr leads to problem in linking the testContVec
+  //  static constexpr sizeType dataOffset() = sizeof(Header) + sizeof(Header)%alignof(T);
+
+  
+  std::unique_ptr<char[]> mPtr;                //! pointer on continuos block containing full object data
+  std::unique_ptr<std::vector<T>> mVecForTree; //! vector for writing objects into the tree
```

Comment:
The mVecForTree is used only for output to TTree. If the latter is not requested, the mVecForTree need not to be initialized. So the alternative to unique_ptr will be the raw pointer (with new/delete). Did I misunderstand you?